### PR TITLE
Fix bool attention mask bias conversion

### PR DESCRIPTION
**Pull request by fallintoplace**
## Summary
- Convert bool attention masks to additive bias with `ISelectLayer` instead of arithmetic involving `-inf`.
- Return `0.0` for valid mask entries and `-inf` for masked entries.
- Add a deterministic bool SDPA mask regression that checks the engine output has no NaN values.

## Testing
- `git diff --check`
- `TORCH_PATH="$PWD/.venv/lib/python3.12/site-packages/torch" bazelisk test //tests/core/conversion/converters:test_scaled_dot_product_attention --test_filter=ATenScaledDotProductAttnMaskBoolDoesNotProduceNaN` was attempted locally. Bazel stayed in analysis without reaching compilation or test execution after about 96 seconds, so the run was interrupted.

### CODE REVIEW

**Comment by zewenli98** [comment]
@fallintoplace Thanks for the contribution. Torchscript is being removed soon (like pytorch 2.14). Please use Dynamo instead.

**Review by zewenli98 — COMMENTED** [review]
> @fallintoplace Thanks for the contribution. Torchscript is being removed soon (like pytorch 2.14). Please use Dynamo instead.

@fallintoplace Pytorch decided not to remove Torchscript support in 2.14. This PR looks good to me. Can you rebase the PR to the main?

**Comment by fallintoplace** [comment]
@zewenli98 Sure, let me push in a few minutes.

**Comment by zewenli98** [comment]
@fallintoplace Can you fix the lint issue? please refer to https://github.com/pytorch/TensorRT/blob/main/CONTRIBUTING.md#coding-guidelines. Thanks
